skiko/src/awtMain/cpp/windows/directXRedrawer.cc

```diff
@@ -53,24 +56,39 @@ class DirectXDevice
     }
 
     void initSwapChain() {
+        RECT windowRect;
+        GetClientRect(window, &windowRect);
+        UINT width = windowRect.right - windowRect.left;
+        UINT height = windowRect.bottom - windowRect.top;
+
         gr_cp<IDXGIFactory4> swapChainFactory4;
         gr_cp<IDXGISwapChain1> swapChain1;
         CreateDXGIFactory2(0, IID_PPV_ARGS(&swapChainFactory4));
         DXGI_SWAP_CHAIN_DESC1 swapChainDesc = {};
-        swapChainDesc.BufferCount = BuffersCount;
+        swapChainDesc.Width = width;
+        swapChainDesc.Height = height;
         swapChainDesc.Format = DXGI_FORMAT_R8G8B8A8_UNORM;
+        swapChainDesc.SampleDesc.Count = 1;
+        swapChainDesc.SampleDesc.Quality = 0;
         swapChainDesc.BufferUsage = DXGI_USAGE_RENDER_TARGET_OUTPUT;
+        swapChainDesc.BufferCount = BuffersCount;
+        swapChainDesc.Scaling = DXGI_SCALING_STRETCH;
         swapChainDesc.SwapEffect = DXGI_SWAP_EFFECT_FLIP_DISCARD;
-        swapChainDesc.SampleDesc.Count = 1;
-        swapChainDesc.Scaling = DXGI_SCALING_NONE;
-        swapChainFactory4->CreateSwapChainForHwnd(queue.get(), window, &swapChainDesc, nullptr, nullptr, &swapChain1);
+        swapChainDesc.AlphaMode = DXGI_ALPHA_MODE_PREMULTIPLIED;
+        swapChainFactory4->CreateSwapChainForComposition(queue.get(), &swapChainDesc, nullptr, &swapChain1);
+
+        IDCompositionDevice* g_pdcDevice = nullptr;
```

Comment:
```suggestion
        DCompositionCreateDevice(0, IID_PPV_ARGS(&g_pdcDevice));
        g_pdcDevice->CreateTargetForHwnd(window, true, &g_pdcTarget);
        g_pdcDevice->CreateVisual(&g_pdcVisual);
        g_pdcVisual->SetContent(swapChain1.get());
        g_pdcTarget->SetRoot(g_pdcVisual.get());
        g_pdcDevice->Commit();
```

and move objects to the fields:
```
    gr_cp<IDCompositionDevice> g_pdcDevice;
    gr_cp<IDCompositionTarget> g_pdcTarget;
    gr_cp<IDCompositionVisual> g_pdcVisual;
```

Otherwise, it seems we have a memory leak.

Reply:
It's just auto-pointer-like objects that holding `nullptr` before setting. I don't see what moving the declaration changes.